### DISCUSSION TITLE
Are CLASSES supported for custom prompt segments?

### DISCUSSION THREAD

**Discussion by aschleifer** [romkatv/powerlevel10k#2734] [General]
Hello, I have a custom prompt segment in my p10k configuration and would like to use CLASSES similar to the `aws` or `kubecontext` segment  in order to use different colors based on the output of the segment.

I have used `typeset -g POWERLEVEL9K_<MY_SEGMENT_NAME>_CLASSES=(...)` and `typeset -g POWERLEVEL9K_<MY_SEGMENT_NAME>_<MY_CLASS_NAME>_BACKGROUND=202` but I don't see any change in the behavior. 

Am I doing something wrong?

**Comment by romkatv** [comment]
CLASSES aren't supported for all segments out of the box. You can either use states (there is an example in `p10k help segment`) or (better), use a plain `if` or `case` within your custom segment and apply conditional formatting.

```zsh
function prompt_my_pwd() {
  local content=$PWD

  case $content/ in
    ~/*)
      # Orange background if the current directory is under $HOME.
      local bg=202
    ;;
    *)
      # Otherwise blue background.
      local bg=27
    ;;
  esac

  p10k segment -b $bg -t "${content//\%/%%}"
}
```

It's almost always easier to format the content of your custom prompt segments directly within the `prompt_` function rather than by using `POWERLEVEL9K_` parameters.
